### PR TITLE
Migrate pallet-scheduler benchmark to v2

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Part of:

- #6202.

### CODE REVIEW

File: substrate/frame/scheduler/src/benchmarking.rs [hunk 1]
```diff
@@ -92,8 +89,9 @@ fn make_task<T: Config>(
 }
 
 fn bounded<T: Config>(len: u32) -> Option<BoundedCallOf<T>> {
-	let call =
-		<<T as Config>::RuntimeCall>::from(SystemCall::remark { remark: vec![0; len as usize] });
+	let call = <<T as Config>::RuntimeCall>::from(frame_system::Call::remark {
```
Comment (re-gius) [inline substrate/frame/scheduler/src/benchmarking.rs:92]: Why did you change `SystemCall` to `frame_system::Call`? I find the old one as readable as the new one, but more compact.

File: substrate/frame/scheduler/src/benchmarking.rs [hunk 2]
```diff
@@ -134,212 +132,293 @@ fn make_origin<T: Config>(signed: bool) -> <T as Config>::PalletsOrigin {
...
-	} verify {
-		//assert_eq!(result, Ok(()));
+		let _result;
+
+		#[block]
+		{
+			_result = Pallet::<T>::service_task(&mut counter, now, now, 0, true, task);
+		}
+
+		// assert!(result.is_ok());
```
Comment (re-gius) [inline substrate/frame/scheduler/src/benchmarking.rs:187]:
```suggestion
		// assert!(_result.is_ok());
```

**Review by re-gius — APPROVED** [review]

File: substrate/frame/scheduler/src/benchmarking.rs [hunk 3]
```diff
@@ -15,27 +15,25 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Scheduler pallet benchmarking.
+//! Pallet pallet benchmarking.
```
Comment (gui1117) [inline substrate/frame/scheduler/src/benchmarking.rs:18]: Why the change?

```suggestion
//! Scheduler pallet benchmarking.
```

File: substrate/frame/scheduler/src/benchmarking.rs [hunk 4]
```diff
@@ -134,132 +132,188 @@ fn make_origin<T: Config>(signed: bool) -> <T as Config>::PalletsOrigin {
...
-	verify {
+		let call = T::Preimages::realize(&make_call::<T>(None))?.0;
+		let result;
+
+		#[block]
+		{
+			result = Pallet::<T>::execute_dispatch(&mut counter, origin, call);
+		}
+
+		assert!(result.is_ok());
```
Comment (gui1117) [inline substrate/frame/scheduler/src/benchmarking.rs:280]: For other reviewer, this has been added by the PR, but I guess it is fine.

File: substrate/frame/scheduler/src/benchmarking.rs [hunk 5]
```diff
@@ -134,132 +132,188 @@ fn make_origin<T: Config>(signed: bool) -> <T as Config>::PalletsOrigin {
...
```
Comment (gui1117) [inline substrate/frame/scheduler/src/benchmarking.rs:262]: For other reviewer, this has been added by the PR, but I guess it is fine.

**Review by gui1117 — APPROVED** [review]
Looks good to me.

The post assertions of those benchmarks are not nice, but it is not related to the PR.

File: substrate/frame/scheduler/src/benchmarking.rs [hunk 6]
```diff
@@ -134,132 +132,188 @@ fn make_origin<T: Config>(signed: bool) -> <T as Config>::PalletsOrigin {
...
-	} verify {
-		//assert_eq!(result, Ok(()));
+		let _result;
+
+		#[block]
+		{
+			_result = Pallet::<T>::service_task(&mut counter, now, now, 0, true, task);
+		}
+
+		// assert!(_result.is_ok());
```
Comment (muharem) [inline substrate/frame/scheduler/src/benchmarking.rs:187]: need to address this comment. more commented code like this below. 
also run benchmarks as we did for democracy pallet

Comment (muharem) [reply substrate/frame/scheduler/src/benchmarking.rs:187]: @AurevoirXavier please open another PR to address this and run the benchmarks before merging. 
See here how to do it: https://github.com/paritytech/polkadot-sdk/pull/6509#issuecomment-2486221769
